src/GogeToken.sol

```diff
@@ -596,346 +584,350 @@ contract DogeGaySon is ERC20, Ownable {
 
     event GasForProcessingUpdated(uint256 indexed newValue, uint256 indexed oldValue);
 
-    event SwapAndLiquify(
-        uint256 tokensSwapped,
-        uint256 ethReceived,
-        uint256 tokensIntoLiqudity
-    );
+    event RoyaltiesTransferred(address indexed wallet, uint256 amountEth);
 
-    event SendDividends(
-        uint256 amount
-    );
-
-    event SwapBNBForTokens(
-        uint256 amountIn,
-        address[] path
-    );
-
-    event ProcessedCakeDividendTracker(
-        uint256 iterations,
-        uint256 claims,
-        uint256 lastProcessedIndex,
-        bool indexed automatic,
-        uint256 gas,
-        address indexed processor
-    );
+    event SwapAndLiquify(uint256 tokensSwapped, uint256 ethReceived, uint256 tokensIntoLiqudity);
+
+    event SendDividends(uint256 amount);
+
+    event BuybackInitiated(uint256 amountIn, address[] path);
+
+    event ProcessedCakeDividendTracker(uint256 iterations, uint256 claims, uint256 lastProcessedIndex, bool indexed automatic, uint256 gas, address indexed processor);
+
+    event Erc20TokenWithdrawn(address token, uint256 amount);
+
+    event AddressExcludedFromCirculatingSupply(address account, bool excluded);
+
+    event Migrated(address indexed account, uint256 amount);
+    
+    event TradingEnabled(uint256 timestamp);
     
     constructor(
         address _marketingWallet,
         address _teamWallet,
-        uint    _totalSupply
+        uint256 _totalSupply,
+        address _gogeV1
 
     ) ERC20("DogeGaySon", "GOGE") {
 
         cakeDividendTracker = new CakeDividendTracker();
 
         marketingWallet = _marketingWallet; //0x4959bCED128E6F056A6ef959D80Bd1fCB7ba7A4B
         teamWallet = _teamWallet; //0xe142E9FCbd9E29C4A65C4979348d76147190a05a
+        devWallet = 0xa13bBda8bE05462232D7Fc4B0aF8f9B57fFf5D02;
         cakeDividendToken = 0x0E09FaBB73Bd3Ade0a17ECC321fD13a19e81cE82;
         
-        GogeV1 = 0xa30D02C5CdB6a76e47EA0D65f369FD39618541Fe;
+        GogeV1 = _gogeV1; //0xa30D02C5CdB6a76e47EA0D65f369FD39618541Fe;
```

Comment:
`GogeV1` can be a constant since this contract is already deployed and doesn't need to be set in the constructor.